web/react/stores/post_store.jsx

```diff
@@ -239,8 +239,15 @@ class PostStoreClass extends EventEmitter {
 
         for (i; i < len; i++) {
             if (postList.posts[postList.order[i]].user_id === userId) {
-                lastPost = postList.posts[postList.order[i]];
-                break;
+                if (rootId) {
+                    if (postList.posts[postList.order[i]].root_id === rootId) {
```

Comment:
Small change that I'd add here would be to also check if the post's id is equal to rootId. As it is, you can't edit the root post of a comment thread from the RHS.